x-pack/plugins/ml/server/shared_services/license_checks/errors.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/* eslint-disable max-classes-per-file */
+
+export class InsufficientFullLicenseError extends Error {
```

Comment:
question: `InsufficientFullLicenseError` and `InsufficientBasicLicenseError` are the same. do we need both classes?

Reply:
i think they are both needed because one is saying that you need at least a basic license and the other is saying you need a platinum or trial license.
You could differentiate using the error message, but I don't see any harm in having separate error classes.